### PR TITLE
RATIS-1440. NettyClientStreamRpc#ChannelInboundHandler add exceptionC…

### PR DESCRIPTION
**Pull request by guohao-rosicky**
NettyClientStreamRpc#ChannelInboundHandler add exceptionCaught method

Used to catch other exceptions.

jira: https://issues.apache.org/jira/browse/RATIS-1440

### CODE REVIEW

**Comment by guohao-rosicky** [comment]
hi @szetszwo  , can you help review this patch?

File: ratis-netty/src/main/java/org/apache/ratis/netty/client/NettyClientStreamRpc.java [hunk 1]
```diff
@@ -123,6 +123,13 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) {
             .map(Queue::poll)
             .ifPresent(f -> f.complete(reply));
       }
+
+      @Override
+      public void exceptionCaught(ChannelHandlerContext ctx,
+                                  Throwable cause) {
+        LOG.warn("netty stream client handler error", cause);
```
Comment (szetszwo) [inline ratis-netty/src/main/java/org/apache/ratis/netty/client/NettyClientStreamRpc.java:130]: Let's include the name as below.
```
        LOG.warn(name + ": exceptionCaught", cause);
```

File: ratis-netty/src/main/java/org/apache/ratis/netty/client/NettyClientStreamRpc.java [hunk 2]
```diff
@@ -123,6 +123,13 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) {
             .map(Queue::poll)
             .ifPresent(f -> f.complete(reply));
       }
+
+      @Override
+      public void exceptionCaught(ChannelHandlerContext ctx,
+                                  Throwable cause) {
```
Comment (szetszwo) [inline ratis-netty/src/main/java/org/apache/ratis/netty/client/NettyClientStreamRpc.java:129]: Please use one line.

**Review by szetszwo — COMMENTED** [review]
@guohao-rosicky , thanks for working on this.  Two minor comments inlined.

**Comment by guohao-rosicky** [comment]
@szetszwo Has been modified,  Please Take A Look.

**Review by szetszwo — APPROVED** [review]
+1 the change looks good.

**Pull request merged**